src/context/AppProvider/AppContext.tsx

```diff
@@ -114,31 +118,56 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
   useEffect(() => {
     if (!wallet) return
     ;(async () => {
-      try {
-        const accountNumber = 0
-        const chainIds = supportedChains
-        const accountMetadataByAccountId = await deriveAccountIdsAndMetadata({
-          accountNumber,
-          chainIds,
-          wallet,
+      let chainIds = Array.from(supportedChains).filter(chainId =>
+        walletSupportsChain({ chainId, wallet }),
+      )
+      const isMultiAccountWallet = wallet.supportsBip44Accounts()
+      for (let accountNumber = 0; chainIds.length > 0; accountNumber++) {
+        // only some wallets support multi account
+        if (accountNumber > 0 && !isMultiAccountWallet) break
+        const input = { accountNumber, chainIds, wallet }
+        const accountMetadataByAccountId = await deriveAccountIdsAndMetadata(input)
+        const accountIds: AccountId[] = Object.keys(accountMetadataByAccountId)
+        const { getAccount } = portfolioApi.endpoints
+        const opts = { forceRefetch: true }
+        const promises = accountIds.map(async id => dispatch(getAccount.initiate(id, opts)))
+
+        const results = await Promise.allSettled(promises)
```

Comment:
Something like this would be more legible but also fine, this whole module reads well and l.140 makes it clear what this refers to
```suggestion
        const accountBalanceByChainId = await Promise.allSettled(promises)
```

Reply:
Looking back into this, since this is an `allSettled()` call, I guess, `settledXYZ` would make more sense?